### PR TITLE
Implement delete/update repository API

### PR DESCRIPTION
**Pull request by ywk253100**
Implement delete/update repository API

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

**Comment by ywk253100** [comment]
Split `Listxxx()(total int64, xxx []model, err error)` into two methods: `LIstxxx()(xxx []model, err error)` and `Count()(total int64, err error)`

File: src/server/v2.0/handler/repository.go [hunk 1]
```diff
@@ -18,20 +18,99 @@ import (
...
+	if params.Name != nil {
+		query.Keywords["Name"] = *(params.Name)
+	}
+	if params.Page != nil {
+		query.PageNumber = *(params.Page)
+	}
+	if params.PageSize != nil {
+		query.PageSize = *(params.PageSize)
+	}
+	total, err := r.repoCtl.Count(ctx, query)
```
Comment (wy65701436) [inline src/server/v2.0/handler/repository.go]: is it needed?

File: api/v2.0/swagger.yaml [hunk 2]
```diff
@@ -16,6 +16,94 @@ securityDefinitions:
...
+              $ref: '#/definitions/Repository'
+        '401':
+          $ref: '#/responses/401'
+        '403':
+          $ref: '#/responses/403'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
+  /projects/{project_name}/repositories/{repository_name}:
```
Comment (wy65701436) [inline api/v2.0/swagger.yaml]: 400?

Comment (ywk253100) [reply api/v2.0/swagger.yaml]: added

**Comment by ywk253100** [comment]
> lgtm, but where do we use the update repository?

Update the description of repository

**Pull request merged**